code/solr/SolrSearchAdapter.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+class SolrSearchAdapter implements SearchAdapterInterface
+{
+    /**
+     * @param SearchCriterion $criterion
+     * @return string
+     */
+    public function generateQueryString(SearchCriterion $criterion)
+    {
+        $writer = $this->getSearchQueryWriter($criterion);
+
+        return $writer->generateQueryString($criterion);
+    }
+
+    /**
+     * @param string $conjunction
+     * @return string
+     * @throws InvalidArgumentException
+     */
+    public function getConjunctionFor($conjunction)
+    {
+        switch ($conjunction) {
+            case SearchCriteria::UND:
+                return ' AND ';
+            case SearchCriteria::ODER:
+                return ' OR ';
```

Comment:
This is duplicating the constant values. How would you feel about this instead?
```php
case SearchCriteria::UND:
case SearchCriteria::ODER:
    return sprintf(' %s ', $conjunction);
```

Note: I commented earlier about the naming of these constants